pkg/kubelet/kubelet_pods.go

```diff
@@ -135,7 +135,31 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			return nil, err
 		}
 		if mount.SubPath != "" {
+			fileinfo, err := os.Lstat(hostPath)
+			if err != nil {
+				return nil, err
+			}
+			perm := fileinfo.Mode()
+
 			hostPath = filepath.Join(hostPath, mount.SubPath)
+
+			if err := os.MkdirAll(hostPath, perm); err != nil {
```

Comment:
This could really use a comment - it is subtle - explain why we're checking the source -- it might be auto-created when referenced, and we need it to have the right ownership and mode.